cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/repo.go

```diff
@@ -48,21 +48,40 @@ var (
 )
 
 // namespace maybe apiv1.NamespaceAll, in which case repositories from all namespaces are returned
+// the repositories to which the caller context has no access are filtered out
 func (s *Server) listReposInNamespace(ctx context.Context, namespace string) ([]sourcev1.HelmRepository, error) {
-	client, err := s.getClient(ctx, namespace)
```

Comment:
I assume this would return a client that was either for the specific namespace, or for all namespaces? But the new code below seems to just ignore `namespace`... fetching all repos in all namespaces and returning those the user has access to. Don't we only want to do that if `namespace = apiv1.NamespaceAll`, as per the function comment above?